Move examples from cog-examples to cog (#3055)
replicate/cog: examples/experimental/resnet-managed-weights/README.md

**`source.uri`** -- where the weights come from. Two formats:

- `hf://<org>/<repo>` -- pulls from HuggingFace Hub
- A local directory path (e.g. `weights/`) -- uses files already on disk

**`source.exclude`** -- glob patterns for files to skip. Most HF repos ship
weights in multiple formats (PyTorch, TF, Flax, ONNX). Exclude the ones you
don't need -- it'll save gigabytes.

**`target`** -- the absolute path where weight files land inside the container.
Your `run.py` loads from this path. Must start with `/`.

## Getting started

### 1. Import weights

This downloads weight files from HuggingFace into the local cache and
generates `weights.lock`:

```sh
cd examples/experimental/resnet-managed-weights
cog weights import
```

The lockfile records digests and sizes for every file. It's how cog knows
whether weights have changed on subsequent imports. Commit `weights.lock`
to version control.

### 2. Run a prediction locally

```sh
cog run -i image=@hotdog.png
```

Locally, cog assembles the weight files from the cache and bind-mounts them
into the container at the `target` path. You don't need to push anything to test.

### 3. Build and push

```sh
cog push
```

> This builds the model image and pushes it to the registry specified by `image:`

Reviewer: This still refers to the old `image:` field, but this PR changed the example config to use `model:`.

```suggestion
This builds the model image and pushes it as the model named by `model:`
```

Author: Fixed — now says it pushes the model named by `model:`.

in `cog.yaml`, alongside the weight layers as an OCI image index. The weights
and model image are separate artifacts in the registry -- the image index ties
them together.

## Important: `.dockerignore`

The `.dockerignore` excludes `weights/` and `.cog/weights-cache/` from the
Docker build context. This matters if you're using local directory weight
sources -- without it, Docker would send the full weight directory to the
build daemon on every `cog build`.

## Adapting this for your own model

1. Copy this directory as a starting point
2. Edit `cog.yaml`:
   - Change `source.uri` to your HuggingFace repo (or a local path)
   - Adjust `exclude` patterns for the formats you don't need
   - Set `target` to wherever your code expects to find the weights

>    - Set `image` to your registry destination (required for `cog push`)

Reviewer: Same stale `image:` guidance here; users following this would edit a field the example no longer uses.

```suggestion
   - Set `model` to your model name (required for `cog push`)
```

Author: Fixed — now instructs setting `model` to your model name.
